sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -88,17 +88,19 @@ public DefaultRedirectStrategy(int maxAttempts, String locationHeader, Set<HttpM
     public boolean shouldAttemptRedirect(HttpPipelineCallContext context,
                                          HttpResponse httpResponse, int tryCount,
                                          Set<String> attemptedRedirectUrls) {
-        String redirectUrl = tryGetRedirectHeader(httpResponse.getHeaders(), getLocationHeader());
 
-        if (isValidRedirectCount(tryCount)
-            && redirectUrl != null
-            && !alreadyAttemptedRedirectUrl(redirectUrl, attemptedRedirectUrls)
-            && isValidRedirectStatusCode(httpResponse.getStatusCode())
+        if (isValidRedirectStatusCode(httpResponse.getStatusCode())
```

Comment:
To reduce nesting, you can check for (!isValidRedirectCount || !isAllowedRedirectMethod) then return false. That will allow you to remove the nested if statement.

Not Blocking